src/main/java/org/terasology/magicalstats/event/BeforeDrainedEvent.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.magicalstats.event;
+
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.AbstractConsumableValueModifiableEvent;
+import org.terasology.entitySystem.prefab.Prefab;
+
+
+public class BeforeDrainedEvent extends AbstractConsumableValueModifiableEvent {
```

Comment:
Javadoc on Events, Systems, Components etc is hugely appreciated :D Even if the names should be fairly obvious it is nice with a quick little sentence to spell stuff out (later that will likely be used to auto-generate modding API docs)